net/mwan3/files/usr/sbin/mwan3

```diff
@@ -61,13 +61,18 @@ ifup()
 		exit 0
 	}
 
-	json_load $(ubus -S call network.interface.$1 status)
-	json_get_vars up l3_device
+	status=$(ubus -S call network.interface.$1 status)
+	[ -n "$status" ] && {
+		json_load $status
+		json_get_vars up l3_device 
+	}
+
 	config_get enabled "$1" enabled 0
 
-	if [ "$up" -eq 1 ] \
+
+	if [ "$up" == "1" ] \
 		&& [ -n "$l3_device" ] \
-		&& [ "$enabled" -eq 1 ]; then
+		&& [ "$enabled" == "1" ]; then
```

Comment:
The same her
Why did you change the check from '-eq' to '=='?

Reply:
The various enabled with double quotes is string type, while eq is compared with two integer, I think  that using '==' is more reasonable for comparison two string type, otherwise, if the length of string is zero, it causes *out of range* error.

Reply:
@feckert  Yes, we change *==* to *=*.